pandas/core/categorical.py

```diff
@@ -356,8 +357,26 @@ def itemsize(self):
         """ return the size of a single category """
         return self.categories.itemsize
 
-    def reshape(self, new_shape, **kwargs):
-        """ compat with .reshape """
+    def reshape(self, new_shape, *args, **kwargs):
+        """
+        An ndarray-compatible method that returns
+        `self` because categorical instances cannot
+        actually be reshaped.
+        """
+        # while the 'new_shape' parameter has no effect,
+        # we should still enforce valid shape parameters
+        shape = np.asarray(new_shape)
+
+        if not com.is_integer_dtype(shape.dtype):
+            raise TypeError("an integer is required")
+
+        if shape.ndim > 1:
+            raise TypeError("'new_shape' can be at most one-dimensional")
+
```

Comment:
Search of `test_reshape` in `pandas/test/test_categorical`, and you'll see that ALL of these conditions are tested.